org.eclipse.m2e.profiles.core/src/org/eclipse/m2e/profiles/core/internal/MavenProfilesCoreActivator.java

```diff
@@ -10,83 +10,59 @@
+import org.osgi.framework.BundleActivator;
 import org.osgi.framework.BundleContext;
+import org.osgi.util.tracker.ServiceTracker;
+
 
 /**
  * The activator class controls the plug-in life cycle
  *
  * @author Fred Bricon
  * @since 1.5.0
  */
-public class MavenProfilesCoreActivator extends Plugin {
-
-	// The plug-in ID
-	public static final String PLUGIN_ID = "org.eclipse.m2e.profiles.core.internal"; //$NON-NLS-1$
-
-	private IProfileManager profileManager;
-
-	// The shared instance
-	private static MavenProfilesCoreActivator plugin;
-
-	/**
-	 * The constructor
-	 */
-	public MavenProfilesCoreActivator() {
-	}
-
-	/*
-	 * (non-Javadoc)
-	 * @see org.eclipse.ui.plugin.AbstractUIPlugin#start(org.osgi.framework.BundleContext)
-	 */
-	public void start(BundleContext context) throws Exception {
-		super.start(context);
-		plugin = this;
-		profileManager = new ProfileManager();
-	}
+public class MavenProfilesCoreActivator implements BundleActivator {
 
-	/*
-	 * (non-Javadoc)
-	 * @see org.eclipse.ui.plugin.AbstractUIPlugin#stop(org.osgi.framework.BundleContext)
-	 */
-	public void stop(BundleContext context) throws Exception {
-		plugin = null;
-		super.stop(context);
-	}
+  private static MavenProfilesCoreActivator instance;
 
-	/**
-	 * Returns the shared instance
-	 *
-	 * @return the shared instance
-	 */
-	public static MavenProfilesCoreActivator getDefault() {
-		return plugin;
-	}
+  private ServiceTracker<IProfileManager, IProfileManager> profileManagerTracker;
 
-	public static IStatus getStatus(String message) {
-		return new Status(IStatus.ERROR, PLUGIN_ID, message);
-	}
+  private BundleContext context;
 
-	public static IStatus getStatus(String message, Throwable e) {
-		return new Status(IStatus.ERROR, PLUGIN_ID, message,e);
-	}
+  /*
+   * (non-Javadoc)
+   * @see org.eclipse.ui.plugin.AbstractUIPlugin#start(org.osgi.framework.BundleContext)
+   */
+  public void start(BundleContext context) throws Exception {
+    this.context = context;
+    instance = this;
+  }
 
-	public static void log(Throwable e) {
-		IStatus status = new Status(IStatus.ERROR, PLUGIN_ID, e.getLocalizedMessage(), e);
-		getDefault().getLog().log(status);
-	}
+  /*
+   * (non-Javadoc)
+   * @see org.eclipse.ui.plugin.AbstractUIPlugin#stop(org.osgi.framework.BundleContext)
+   */
+  public void stop(BundleContext context) throws Exception {
+    instance = null;
+  }
 
-	public static void log(String message) {
-		IStatus status = new Status(IStatus.ERROR, PLUGIN_ID, message);
-		getDefault().getLog().log(status);
-	}
+  /**
+   * Returns the shared instance
+   *
+   * @return the shared instance
+   */
+  public static MavenProfilesCoreActivator getDefault() {
+    return instance;
+  }
 
-	public IProfileManager getProfileManager() {
-		return profileManager;
-	}
+  @Deprecated(forRemoval = true) //only used for test code at the moment...
+  public synchronized IProfileManager getProfileManager() {
```

Comment:
Are you planning to update the tests accordingly to use the service from the registry? Then we could entirely remove this activator.
You could do it like in your last change (and Mickael waits until the .gitmodules change is reverted before this is merged :) 😄).

Reply:
My idea ways that I first merge this, then adjust the test, then remove the activator, its a bit more steps but I think better than all theses hassle with .gitmodules changes ...

Reply:
Sure, that's also possible. As you prefer. 👍🏽